framework/src/materials/DerivativeParsedMaterialHelper.C

```diff
@@ -101,14 +101,16 @@ DerivativeParsedMaterialHelper::assembleDerivatives()
     }
 
     // copy coupled material properties
-    for (unsigned int i = 0; i < master->_mat_prop_descriptors.size(); ++i)
+    auto start = _mat_prop_descriptors.size();
+    for (unsigned int i = start; i < master->_mat_prop_descriptors.size(); ++i)
```

Comment:
OK - but kind of scatterbrained. `auto` followed immediately by `unsigned int`!  If you are going to do this, why not `decltype`? or even better use our macro `beginIndex()`!

Reply:
`beginIndex`?! That would get me `0` which is wrong. I'll do the decltype dance.

Reply:
Nope `beginIndex` takes multiple paramters, there's this form too:
`beginIndex(container, start_index)`

Reply:
Oh, there is a two argument version of that macro...

Reply:
So I'm not insisting you use that method (I personally like it), but most people prefer to just declare the wrong type directly instead (e.g. `unsigned int`).